eng/scripts/ValidateProjectCoverage.ps1

```diff
@@ -0,0 +1,199 @@
+﻿#!/usr/bin/env pwsh
+<#
+.SYNOPSIS
+    Validates the code coverage policy for each project.
+.DESCRIPTION
+    This script compares code coverage with thresholds given in "MinCodeCoverage" property in each project.
+    The script writes an error for each project that does not comply with the policy.
+.PARAMETER CoberturaReportXml
+    Path to the XML file to read the code coverage report from in Cobertura format
+.EXAMPLE
+    PS> .\ValidatePerProjectCoverage.ps1 -CoberturaReportXml .\Cobertura.xml
+#>
+
+param (
+    [Parameter(Mandatory = $true, HelpMessage="Path to the XML file to read the code coverage report from")]
+    [string]$CoberturaReportXml
+)
+
+function Write-Header {
+    param($message, [bool]$isError);
+    $color = if ($isError) { 'Red' } else { 'Green' };
+    Write-Host $message -ForegroundColor $color;
+    Write-Host ("=" * 80)
+ }
+function Get-XmlValue { param($X, $Y); return $X.SelectSingleNode($Y).'#text' }
+
+Write-Verbose "Reading cobertura report..."
+[xml]$CoberturaReport = Get-Content $CoberturaReportXml
+if ($null -eq $CoberturaReport.coverage -or 
+    $null -eq $CoberturaReport.coverage.packages -or
+    $null -eq $CoberturaReport.coverage.packages.package -or 
+    0 -eq $CoberturaReport.coverage.packages.package.count)
+{
+    return
+}
+
+$ProjectToMinCoverageMap = @{}
+
+Get-ChildItem -Path src -Include '*.*sproj' -Recurse | ForEach-Object {
+    if ($_.FullName.Contains('Aspire.ProjectTemplates')) {
+        return
+    }
+
+    $XmlDoc = [xml](Get-Content $_)
```

Comment:
super nit: maybe `projectDoc` or something to indicate that this is the msbuild project file.